packages/amplify-graphql-index-transformer/src/resolvers/resolvers.ts

```diff
@@ -851,22 +910,26 @@ export const generateAuthExpressionForSandboxMode = (enabled: boolean): string =
   );
 };
 
-export function getDBInfo(ctx: TransformerContextProvider, modelName: string) {
+export const getDBInfo = (ctx: TransformerContextProvider, modelName: string): DataSourceType => {
   const dbInfo = ctx.modelToDatasourceMap.get(modelName);
-  const result = dbInfo ?? { dbType: DDB_DB_TYPE, provisionDB: true };
+  const result = dbInfo ?? { dbType: DDB_DB_TYPE, provisionDB: true, provisionStrategy: DynamoDBProvisionStrategy.DEFAULT };
   return result;
-}
+};
 
-export function getDBType(ctx: TransformerContextProvider, modelName: string) {
+export const getDBType = (ctx: TransformerContextProvider, modelName: string): DBType => {
   const dbInfo = getDBInfo(ctx, modelName);
   const dbType = dbInfo ? dbInfo.dbType : DDB_DB_TYPE;
   return dbType;
-}
+};
 
 export const getVTLGenerator = (dbInfo: DataSourceType | undefined): RDSIndexVTLGenerator | DynamoDBIndexVTLGenerator => {
   const dbType = dbInfo ? dbInfo.dbType : DDB_DB_TYPE;
   if (dbType === DDB_DB_TYPE) {
     return new DynamoDBIndexVTLGenerator();
   }
+
+  if (dbType === 'MySQL') {
+    return new RDSIndexVTLGenerator();
+  }
```

Comment:
This needs to be updated as per https://github.com/aws-amplify/amplify-category-api/blob/b3905a62e16ff56287ac0884fe7881780b2903cf/packages/amplify-graphql-index-transformer/src/resolvers/resolvers.ts#L866